### PR TITLE
fix(fa): Fixing error messages in financial-aid 

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# 🆙 
## What

- Adding more detailed error messages for api 
- Adding more detail also for formatting dates 

## Why

- The feature was asked about 

## Screenshots / Gifs
Screenshot 2024-08-06 at 14 00 23
Screenshot 2024-08-06 at 14 04 04
Screenshot 2024-08-06 at 14 04 15
Screenshot 2024-08-06 at 14 04 32
Screenshot 2024-08-06 at 14 01 24

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for invalid API keys and municipality codes in user service and application controller.
	- Clearer validation messages for missing headers in API requests.
	- Introduced date validation utility to ensure correct date formats and logical constraints.
  
- **Bug Fixes**
	- Improved checks for empty arrays in API key searches to prevent unnecessary processing.

### CODE REVIEW

File: apps/financial-aid/open-api/src/app/app.controller.ts [hunk 1]
```diff
@@ -26,6 +33,16 @@ export class AppController {
...
   ): Promise<ApplicationModel[]> {
+    if (!apiKey && !municipalityCode) {
+      throw new BadRequestException('API-Key and Municipality-Code are missing')
+    }
+    if (!apiKey) {
+      throw new BadRequestException('API-Key is missing')
+    }
+    if (!municipalityCode) {
+      throw new BadRequestException('Municipality-Code is missing')
+    }
```
Comment (Rikkiabb) [inline apps/financial-aid/open-api/src/app/app.controller.ts:44]: Should this logic be moved to a guard?

File: apps/financial-aid/open-api/src/app/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { BadRequestException } from '@nestjs/common'
+
+export const isDateValid = (startDate: string, isEndDate: boolean): boolean => {
```
Comment (Rikkiabb) [inline apps/financial-aid/open-api/src/app/helpers.ts:3]: Should isEndDate rather be a string or a enum?

**Review by Rikkiabb — APPROVED** [review]
Two things ✌️

File: apps/financial-aid/open-api/src/app/app.controller.ts (hunk 1 above)
Comment (MargretFinnboga) [reply apps/financial-aid/open-api/src/app/app.controller.ts]: ofc! 🤩

**Pull request merged**